dbms/src/Storages/MergeTree/IMergeTreeDataPart.cpp

```diff
@@ -768,6 +771,8 @@ void IMergeTreeDataPart::remove() const
                                        ErrorCodes::CANNOT_UNLINK);
         }
 
+        unlink((to + "/" + DELETE_ON_DESTROY_MARKER_PATH).c_str());
```

Comment:
We still must check the error code, check that it matches our expectations (success or file does not exist) and throw exception otherwise.

Reply:
+